code/__DEFINES/maths.dm

```diff
@@ -237,6 +237,13 @@ proc/leftmost_bit(num)
 			pos++
 	return pos
 
+// inverts a dir within a pair of opposites
+#define INVERSEONEDIR(startdir, axis) ((startdir & (axis))^(((((((startdir&(axis))<<1)&(axis)))^((((startdir&(axis)))&(axis<<1))))) * 1.5))
+// inverts a dir completely
+#define INVERSE_DIR(startdir) INVERSEONEDIR(startdir, SOUTH+NORTH) | INVERSEONEDIR(startdir, EAST+WEST) | INVERSEONEDIR(startdir, UP+DOWN)
+// inverts a dir on both east/west and south/north axis
+#define INVERSE_FLAT_DIR(startdir) INVERSEONEDIR(startdir, SOUTH+NORTH) | INVERSEONEDIR(startdir, EAST+WEST)
```

Comment:
Did you intentionally make this as ugly and unreadable as possible?
It definitely doesn't have to look like _that_, not to mention that `turn(dir, angle)` would do the trick perfectly fine.

Reply:
turn doesn't work vertically, but the formatting issues I can fix, and inverse_flat_dir doesn't need to exist.

Reply:
never mind, it does need to exist

Reply:
or does it?